WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingTaskViewModel.swift

```diff
@@ -7,45 +7,53 @@ struct StoreOnboardingTaskViewModel: Identifiable, Equatable {
     let icon: UIImage
     let title: String
     let subtitle: String
+    let badgeText: String?
 
     var isComplete: Bool {
         task.isComplete
     }
 
-    init(task: StoreOnboardingTask) {
+    init(task: StoreOnboardingTask, isEligibleForProductDescriptionAI: Bool) {
```

Comment:
Nit: `isEligibleForProductDescriptionAI` seems to be irrelevant to task types other than `.addFirstProduct`. Should we make this an associated value of that type, or move the check to this class instead?

Reply:
> Should we make this an associated value of that type

The task type `TaskType` is directly decoded from the network response from `init(from:)` with a decoder, it doesn't seem straightforward to pass the condition there.

> or move the check to this class instead

If I understood this correctly, did you mean instantiating `ProductFormAIEligibilityChecker` within the view model? If so, then we need to pass `featureFlagService` and `Site` to the view model, which exposes the implementation detail of the eligibility checker.

Because the store onboarding naturally doesn't have a direct relationship with product form AI, this knowledge of product description AI eligibility has to be integrated with store onboarding somehow. Lemme know if you think I misunderstood something / any other suggested approaches.

Reply:
Thanks for the explanation! I understand it's not straightforward to inject `isEligibleForProductDescriptionAI` in another way. Another idea I can think of is to inject `badgeText` directly into the task view model with the default `nil` value - does that sound better?

Reply:
Sure, I made some updates in https://github.com/woocommerce/woocommerce-ios/pull/9659/commits/b13a6c6e9468e683273426b4edc16b5f1b32b5d3 - I kept the localization string constant in the task view model just to have all strings in one place.